src/Notification.tsx

```diff
@@ -239,20 +239,35 @@ class Notification extends Component<NotificationProps, NotificationState> {
 Notification.newInstance = function newNotificationInstance(properties, callback) {
   const { getContainer, ...props } = properties || {};
   const div = document.createElement('div');
+  let mountTarget = document.body;
   if (getContainer) {
-    const root = getContainer();
-    root.appendChild(div);
+    mountTarget = getContainer();
+    mountTarget.appendChild(div);
   } else {
-    document.body.appendChild(div);
+    mountTarget.appendChild(div);
   }
   let called = false;
+  function isChildOfDocument(child) {
+    let parentNode = child.parentNode;
+    while (parentNode) {
+      if (document === parentNode) {
+        return true;
+      }
+      parentNode = parentNode.parentNode;
+    }
+    return false;
+  }
   function ref(notification: Notification) {
     if (called) {
       return;
     }
     called = true;
     callback({
       notice(noticeProps) {
+        if (getContainer && (!mountTarget || !isChildOfDocument(mountTarget))) {
+          mountTarget = getContainer();
+          mountTarget.appendChild(div);
```

Comment:
Seems this code is same as above, could we move into a function instead?

Reply:
👌，please review